File: 2019_four.c
//
//  2019_four.c
//  
//
//  Created by Eli Kurtz on 12/3/19.
//

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <string.h>

#define LOW 387638
#define HIGH 919123


//plan of attack: generate all numbers between range and then eliminate

//int remove_decrease (array, c) {
    //forgot that I couldnt easily index an integer in C. contemplating what to do now. maybe I will creat a list of all those numbers that do not satisfy the decrease and then remove them from the first list?
    
    //int false_array[c][6]
    
    
    
    //for(i=0; i < n; i++) {
        
        
        
    //}
    
    
//}



int main () {
    
    int a = LOW;
    char buffer[6];
    int b = HIGH;
    int c = b - a;
    int i = 0;
    
    int array[c];
    
    while (a < b) {
        
        a = itoa(a);
        array[i] = a;
        printf("\n%s\n", array[i]);
        i++;
        a++;
        
    }
    
    //https://stackoverflow.com/questions/10162465/implicit-declaration-of-function-itoa-is-invalid-in-c99
    
}

File: 2019_four.py
def doubles(array, n):
    x = 0
    clean_array = []
    a = 0
    b = 1

    while x < n:
        if array[x][a] == array[x][b]:
            a += 1
            b += 1
            clean_array.append(array[x])
            print("hello")
            if b == 6:
                break
        x += 1

    return clean_array


def remove_decrease(array, n):

    x = 0
    clean_array = []
    while x < n:
        if array[x][0] <= array[x][1] and array[x][1] <= array[x][2] and array[x][2] <= array[x][3] and \
        array[x][3] <= array[x][4] and array[x][4] <= array[x][5]:
            clean_array.append(array[x])
        x += 1

    return(clean_array)


def array_it ():

    possible_password = []
    low = 387638
    high = 919123
    password_range = high - low
    a = low
    b = high
    i = 0

    while a < b:

        x = list(map(int, str(a)))
        possible_password.append(x)
        i += 1
        a += 1

    possible_password = remove_decrease(possible_password, password_range)
    possible_password = doubles(possible_password, password_range)

    print(possible_password)

array_it()

